### PR TITLE
IMP: Add read_orientation param to extract_reads

### LINKED ISSUE #124: `extract-reads`: add read orientation parameter
**Improvement Description**
`extract-reads` should have a parameter to specify the orientation of reads (forward | reverse | both) or just autodetect the orientation by attempting to match primers to the reverse complement of reads that do not match on first pass.

**Current Behavior**
Assumes all reads are forward orientation. If databases are in mixed orientation, this can cause trouble (see forum xref below)

**Proposed Behavior**
Add parameter to specify read orientation. If read orientation is `both`, reads that do not match primers will be reverse complemented to attempt a second extraction.

*This will have the added benefit of providing a way to homogenize mixed-orientation reads, which currently cause problems with `classify-sklearn`.*

**References**
[forum xref](https://forum.qiime2.org/t/multiple-orientation-repseqs-not-fastq-feature-classifier-extract-reads-modifier/7822)

### PR DESCRIPTION
**Pull request by Oddant1**
Closes #124. Adds a basic implementation of the `read_orientation` parameter to `extract_reads` that defaults to checking both forward and reverse orientation reads

### CODE REVIEW

**Comment by Oddant1** [comment]
I appear to have broken the commit while fixing the merge conflict? I'll look at it

**Comment by Oddant1** [comment]
I did some benchmarks with a reverse complimented greengenes database and the results are. . . interesting.

Selection_096

I ran the command with both, then only reverse, and according the the `time` command, if anything it takes LONGER to run with only reverse. Those four runs of the command alternate "both, reverse, both, reverse"

**Comment by nbokulich** [comment]
that's a bit strange. greengenes might actually be in mixed orientation
(afaik they are supposed to be forward orientation but maybe not...)

**Comment by Oddant1** [comment]
So how do we want to proceed from here then?

**Comment by nbokulich** [comment]
If the read orientation parameter is not helping us then I feel like we should just close this out rather than add a step that will increase runtime without obvious benefits. 

Sorry you've put in the work and we are at a dead end... if you want you could give one more try to make sure the greengenes reads really are in the same orientation before repeating the test (a small set of seqs should be fine for this).

@BenKaehler @thermokarst any thoughts about this?

**Comment by Oddant1** [comment]
I figured I should go look at the qzas actually produced by `both` and `reverse` and the .fasta file in the `both` qza is 1.6gigs while the `reverse` is 1.3. This makes me think the reads are in mixed orientation

**Comment by thermokarst** [comment]
I'll try to take a look at this in the next day or two.

**Comment by Oddant1** [comment]
Drive by ping @thermokarst. Are you still planning on looking at this?

**Comment by thermokarst** [comment]
Hey @Oddant1, @David-Rod is going to take a pass through reviewing this PR, when he is done I will do a final check. Thanks!

File: q2_feature_classifier/tests/data/dna-sequences-reverse.fasta [hunk 1]
```diff
@@ -0,0 +1,10 @@
+>Sequence1
+CCTGCACGTTCTCT
+>Sequence2
+CCTGCACTTTCTCT
+>Sequence3
+CCTGCAGGTTCTCT
+>Sequence4
+CCTGCCCGTTCTCT
+>Sequence5
+CCTGCAAGTTCTCT
```
Comment (David-Rod) [inline q2_feature_classifier/tests/data/dna-sequences-reverse.fasta]: @Oddant1 please add new line to this file.

**Comment by thermokarst** [comment]
Hey @nbokulich, can you take a look at the tests (and test data), just to make sure this is behaving as expected (and being tested appropriately). Not sure what Coveralls is complaining about, btw --- the test suite didn't "uncover" any previously covered lines (that's what the detail page says, anyway)...

**Pull request merged**